src/source-pcap.c

```diff
@@ -129,14 +151,50 @@ void TmModuleDecodePcapRegister (void)
     tmm_modules[TMM_DECODEPCAP].flags = TM_FLAG_DECODE_TM;
 }
 
+/**
+ * \brief Update 64 bit |last| value from |current32| value taking one
+ * wrap-around into account.
+ */
+static inline void UpdatePcapStatsValue64(uint64_t *last, uint32_t current32)
+{
+    uint32_t last32 = *last; /* slice lower 32bits */
```

Comment:
I don't know enough about defined/undefined behavior to judge this. Would it make sense to first do a `uint64_t`with a mask that just leaves the lower 32 bits? Then downcast it?

Reply:
unsigned to unsigned conversion (across different sizes) is well defined in the C standard.

As ```last``` is an uint64_t, it simply slices, i.e. it essentially "masks and downcasts" by default.

Reply:
FYI, as for what integer conversion is defined, or undefined, or implementation dependent... One of the better practical explanations is: https://unspecified.wordpress.com/2011/08/08/integer-conversions-in-c/

You can always get into more detail through the CERT pages:  https://wiki.sei.cmu.edu/confluence/display/c/INT02-C.+Understand+integer+conversion+rules or the C standard. This allows you to go nuts about promotion, ranks, etc if you fancy that. ;)